package NDoom;

import "kernel/doom/chunked_wad/protos/modulo_doc_chunk_mapping_info.proto";
import "kernel/doom/chunked_wad/protos/sequential_doc_chunk_mapping_info.proto";

message TDocChunkMappingUnion {
    oneof Mapping {
        // Note, that these field numbers are specifically chosen to not overlap with those in TModuloDocChunkMappingInfo
        // Due to backwards compatibility issues, we try to deserialize bytes as TDocChunkMappingUnion first,
        // and then, if the resulting message is empty, we try to interpret them as TModuloDocChunkMappingInfo
        TSequentialDocChunkMappingInfo SequentialMapping = 100;
        TModuloDocChunkMappingInfo ModuloMapping = 101;
    }
}
